frontend/packages/moocfi-auth/index.ts

```diff
@@ -0,0 +1,181 @@
+import axios from "axios"
+import Cookies from "universal-cookie"
+
+/*
+const axios = require("axios")
+const Cookies = require("universal-cookie")
+*/
+
+const BASE_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://mooc.fi"
+    : "http://localhost:4000"
+
+interface ExtraFields {
+  namespace?: string
+  data?: any
+}
+
+interface UserFields {
+  organizational_id?: string
+  first_name?: string
+  last_name?: string
+}
+
+interface CreateFields {
+  email: string
+  password: string
+  confirmPassword: string
+  extra_fields: ExtraFields
+  user_fields: UserFields
+  origin?: string
+  provider?: string
+  language?: string
+  domain?: string
+  priority?: string
+}
+
+export const createUser = async (data: CreateFields) => {
+  const cookies = new Cookies()
+  let origin = data.origin || "mooc.fi"
+  let provider = data.provider || "native"
+  let language = data.language || "en"
+  let domain = data.domain || "localhost"
+  let priority = data.priority || null
+
+  return await axios({
```

Comment:
We have so may API calls inside react views that it would make sense to move the all the axios calls to their own file for the separation of concerns . I would mimic the approach we we're using in the backend, so I would crate a services folder and add a service file for tmc there (like https://github.com/rage/mooc.fi/blob/master/backend/services/tmc.ts)

Reply:
This isn't a react view, but these files are ones with axios calls in their views: 
https://github.com/rage/mooc.fi/blob/auth-sso/frontend/pages/%5Blng%5D/register-completion/%5Bslug%5D/index.tsx
https://github.com/rage/mooc.fi/blob/auth-sso/frontend/pages/%5Blng%5D/authorization.tsx
https://github.com/rage/mooc.fi/blob/auth-sso/frontend/pages/%5Blng%5D/auth/clients.tsx
https://github.com/rage/mooc.fi/blob/auth-sso/frontend/pages/%5Blng%5D/auth/client-details.tsx

Also, why are service calls the go-to solution instead of using a single source of truth with a middleware like Thunk or Sagas?